addons/website_twitter/models/website_twitter.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import logging
+import werkzeug
+from urllib2 import urlopen, Request, HTTPError
+from odoo import api, fields, models
+
+API_ENDPOINT = 'https://api.twitter.com'
+API_VERSION = '1.1'
+REQUEST_TOKEN_URL = '%s/oauth2/token' % API_ENDPOINT
+REQUEST_FAVORITE_LIST_URL = '%s/%s/favorites/list.json' % (API_ENDPOINT, API_VERSION)
+URLOPEN_TIMEOUT = 10
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteTwitter(models.Model):
+    _inherit = 'website'
+
+    twitter_api_key = fields.Char(string='Twitter API key', help='Twitter API Key')
+    twitter_api_secret = fields.Char(string='Twitter API secret', help='Twitter API Secret')
+    twitter_screen_name = fields.Char(string='Get favorites from this screen name')
+
+    @api.model
+    def _request(self, website, url, params=None):
+        """Send an authenticated request to the Twitter API."""
+        access_token = self._get_access_token(website)
+        if params:
+            params = werkzeug.url_encode(params)
+            url = url + '?' + params
+        try:
+            request = Request(url)
+            request.add_header('Authorization', 'Bearer %s' % access_token)
+            return json.load(urlopen(request, timeout=URLOPEN_TIMEOUT))
+        except HTTPError, e:
+            _logger.debug("Twitter API request failed with code: %r, msg: %r, content: %r",
+                          e.code, e.msg, e.fp.read())
+            raise
+
+    @api.model
+    def _refresh_favorite_tweets(self):
+        ''' called by cron job '''
+        website = self.env['website'].search([('twitter_api_key', '!=', False),
+                                          ('twitter_api_secret', '!=', False),
+                                          ('twitter_screen_name', '!=', False)])
+        _logger.debug("Refreshing tweets for website IDs: %r", website.ids)
```

Comment:
@msh-odoo first look the code and then comment , check line no ```56``` in very next file ```website_twitter_config.py```